### PR TITLE
[MAINT] only test label maskers on real atlases on main

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

Changes proposed in this pull request:

- only test label maskers on real atlases on main to minimize CI runs (and also failures when atlases cannot be fetched)

### CODE REVIEW

**Comment by Remi-Gau** [comment]
test failure on pre-release is unrelated: my hunch is that this is due to the last rc1 of sklearn 1.7.0

https://github.com/scikit-learn/scikit-learn/releases/tag/1.7.0rc1
